src/main/java/org/candlepin/subscriptions/tally/InventoryAccountUsageCollector.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.tally;
+
+import org.candlepin.subscriptions.inventory.db.InventoryRepository;
+import org.candlepin.subscriptions.inventory.db.model.InventoryHost;
+import org.candlepin.subscriptions.tally.facts.FactNormalizer;
+import org.candlepin.subscriptions.tally.facts.NormalizedFacts;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * Collects the max values from all accounts in the inventory.
+ */
+public class InventoryAccountUsageCollector {
+
+    private static final Logger log = LoggerFactory.getLogger(InventoryAccountUsageCollector.class);
+
+    private final FactNormalizer factNormalizer;
+    private final InventoryRepository inventoryRepository;
+
+    public InventoryAccountUsageCollector(FactNormalizer factNormalizer,
+        InventoryRepository inventoryRepository) {
+        this.factNormalizer = factNormalizer;
+        this.inventoryRepository = inventoryRepository;
+    }
+
+    @SuppressWarnings("squid:S3776")
+    @Transactional(value = "inventoryTransactionManager", readOnly = true)
+    public Collection<AccountUsageCalculation> collect(Collection<String> products,
+        Collection<String> accounts) {
+        Map<String, AccountUsageCalculation> calcsByAccount = new HashMap<>();
+        try (Stream<InventoryHost> hostStream = inventoryRepository.findByAccountIn(accounts)) {
+            hostStream.forEach(host -> {
+                String account = host.getAccount();
+                if (!calcsByAccount.containsKey(account)) {
```

Comment:
I think this `if` and the line below could be replaced with [putIfAbsent](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#putIfAbsent-K-V-)?

Reply:
DONE